src/coreclr/jit/ifconversion.cpp

```diff
@@ -342,6 +342,7 @@ bool OptIfConversionDsc::IfConvertCheckStmts(BasicBlock* fromBlock, IfConvertOpe
 
                 // These do not need conditional execution.
                 case GT_NOP:
+                case GT_GCPOLL:
```

Comment:
```suggestion
```
This doesn't look right since it will make if-conversion remove these GC polls

Reply:
it doesn't matter in practice since the gc safe points are currently emitted before it, but good for phase ordering resilience